test_gpios.py

```diff
@@ -91,6 +91,12 @@ async def test_interrupts_and_pwm(self, pwm):
             interrupt = self.sw_interrupt
             error_factor = 0.10  # Software PWM can get really inaccurate
 
+        # In order to diagnose a flaky test, we're going to record all tick-related data.
+        should_stop = asyncio.Event()
+        ticks = []
+        tick_stream = await self.board.stream_ticks([interrupt])
```

Comment:
Not sure if this is a useful idea, but it would feel ergonomically simpler if I could have run `interrupt.stream_ticks`, rather than needing to give the board its own interrupt back again. That would only work for streaming from a single interrupt: if you want to stream from multiple ones you'd still need the board. I dunno; something to consider, but not act on yet.